screens/Login.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { ScrollView, StyleSheet, Text, View } from 'react-native';
+import { Colors } from 'react-native/Libraries/NewAppScreen';
+
+import { Button } from '../components/Button';
+import { FormItem } from '../components/FormItem';
+
+export interface LoginState {
+  email: string;
+  password: string;
+  validEmail: boolean;
+  validPassword: boolean;
+}
+
+export class Login extends React.Component<{}, LoginState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      password: "",
+      validEmail: true,
+      validPassword: true
+    };
+  }
+
+  private handleChangeEmail = (email: string) => {
+    this.setState({ email: email });
+  };
+
+  private handleChangePassword = (password: string) => {
+    this.setState({ password: password });
+  };
+
+  private handleButtonPress = () => {
+    var emailRegex = new RegExp("\\w+@\\w+.com");
+    this.setState({ validEmail: emailRegex.test(this.state.email) });
+
+    let regexMinLength = new RegExp(".{7,}");
+    let regexAtLeastADigit = new RegExp(".*\\d.*");
+    let regexAtLeastALetter = new RegExp(".*[a-zA-Z].*");
+    let validPassword = regexMinLength.test(this.state.password)
+      && regexAtLeastADigit.test(this.state.password)
+      && regexAtLeastALetter.test(this.state.password);
+    this.setState({ validPassword: validPassword });
```

Comment:
Alguns pontos sobre seu método:

1. Você está chamando `this.setState()` sempre. Você sabe se o componente renderiza novamente quando você dá um setState "trocando" o valor de um campo para a mesma coisa?
2. Você está chamando o `this.setState()` duas vezes seguidas. Sabe se isso vai fazer com que o componente renderize 2 vezes?

Reply:
[Spoiler sobre a 2](https://medium.com/@wereHamster/beware-react-setstate-is-asynchronous-ce87ef1a9cf3)

Reply:
Conforme conversamos, reuni as chamadas de setState em uma única chamada e também verifico se de fato houve atualização de estado antes de chamar setState